module-2/backend/week_one.md

```diff
@@ -4,31 +4,76 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+
+4.	Why do we follow conventions when creating our actions/path names in our Sinatra routes? 
+Following conventions allows for easier readability and allows for others in the future to maintain and edit the actions/path names in our Sinatra routes.
+
+5.	What types of variables are accessible in our view templates without explicitly passing them?
+Instance variable
+
+6.	Given the following block of code, how would I pass an instance variable count with a value of 1 to my index.erb template?
+get '/horses' do
+	@count = 1
+	name = “Mr. Ed”
+  erb :index, :locals => {:name => name}
+end
+
+7.	In the same code block, how would I pass a local variable name with a value of Mr. Ed? See above answer
+
+8.	What's the purpose of ERB? Embeded Ruby allows us to use html and ruby in the same file.
+
+9.	Why do I need a development AND test database?
+A development database if for all the data. A test database you want clean when running individual tests and want to keep them separate so the test records don’t get mixed up with the actual data.
+
+10.	What's responsive design?
+This just means that a webpage will render differently depending on the width of the browser a user has on different devices. Responsive design is designed with CSS and Javascript.
+
+11.	What is CRUD and why is it important?
+Create, Read, Update, Delete These are four essential actions you need to render on objects in databases.
+
+12.	What does HTTP stand for?
+Hypertest transfer protocol
+
+13.	What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+<%= ruby_code_here %>
+<% ruby_code_here %>
+The first with the equals sign puts the ruby code to the screen. The second without the equals sign only evaluates ruby code.
+
+14.	What's an ORM?
+Object Relational Mapper:  converts data to and from object oriented languages to something the database can understand and vice-versa.
+
+15.	What's the most commonly used ORM?
+SQL? Active Record is used in Ruby.
+
+16.	Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
+GET “/restaurants” : see all restaurants
+GET “/restaurants/:id” see one specific restaurant
+GET “/restaurants/new” see a form in order to create a restaurant
+POST “/restaurants” submit to save and add a new restaurant
+GET “/restaurants/:id/edit” see form to update/edit restaurants
+PUT “/restaurants/:id” submit to update restaurant
+DELETE “/restaurants/:id” delete a specific restaurant
+
+17.	What's a migration?
+It alters a database through a set of instructions.
+
+18.	When you create a migration, does it automatically modify your database?
+No you have to run the migration.
+
+19.	How does a model relate to a database?
+A model sets the associations between the tables and the database. 

```

Comment:
a model directly maps to a table in the database. an instance of a specific model represents one row of the database